lib/elixir/lib/kernel.ex

```diff
@@ -6093,6 +6093,70 @@ defmodule Kernel do
     Macro.compile_apply(mod, fun, [code, options, __CALLER__ | args], __CALLER__)
   end
 
+  @doc """
+  Constructs a millisecond timeout from the given `components`.
+
+  This function is useful for constructing timeouts from human-readable
+  components, to use in functions that expect `t:timeout/0` values
+  (such as `Process.send_after/4` and many others).
+
+  The `components` argument is keyword list which can contain the following
+  keys, each appearing at most once with a non-negative integer value:
+
+    * `:hour` - the number of hours
+    * `:minute` - the number of minutes
+    * `:second` - the number of seconds
+    * `:millisecond` - the number of milliseconds
+
+  The timeout is calculated as the sum of the components, each multiplied by
+  the corresponding factor.
+
+  ## Examples
+
+      iex> to_timeout(hour: 1, minute: 30)
+      5400000
+
+  """
+  @doc since: "1.17.0"
+  @spec to_timeout([{unit, non_neg_integer()}, ...]) :: non_neg_integer()
```

Comment:
We should also support Duration.t, integers and the atom :infinity.

Reply:
Done!